internal/installation/paths.go

```diff
@@ -77,3 +88,65 @@ func LauncherInstallPath() (string, error) {
 	}
 	return defaultSystemInstallPath()
 }
+
+func IsInstallRoot(dir string) bool {
+	return fileutils.FileExists(filepath.Join(dir, InstallDirMarker))
+}
+
+// DetectCorruptedInstallDir will return an error if it detects that the given install path is not a proper
+// State Tool installation path. This mainly covers cases where we are working off of a legacy install of the State
+// Tool or cases where the uninstall was not completed properly.
+func DetectCorruptedInstallDir(path string) error {
+	if !fileutils.TargetExists(path) {
+		return nil
+	}
+
+	isEmpty, err := fileutils.IsEmptyDir(path)
+	if err != nil {
+		return errs.Wrap(err, "Could not check if install dir is empty")
+	}
+	if isEmpty {
+		return nil
+	}
+
+	// Detect if bin dir exists
+	binPath, err := BinPathFromInstallPath(path)
+	if err != nil {
+		return errs.Wrap(err, "Could not detect bin path")
+	}
+	if !fileutils.DirExists(binPath) {
+		return errs.Wrap(ErrCorruptedInstall, "Bin path does not exist: %s", binPath)
+	}
+
+	// Detect if the install dir has files in it
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return errs.Wrap(err, "Could not read directory: %s", path)
+	}
+
+	for _, file := range files {
+		if !file.IsDir() || strings.ToLower(file.Name()) != InstallDirMarker {
```

Comment:
This is probably too aggressive. I think we can get by just checking if it has state executables in the wrong dir.